### PR TITLE
fixing build procedure for macos

### PR DESCRIPTION
**Pull request by jackyalbo**
Signed-off-by: jackyalbo <jacky.albo@gmail.com>

### Explain the changes
1.  Added syscall implementation for macos. source: https://github.com/streamich/libsys/blob/master/syscall/darwin.c
2. SYS_setresuid and SYS_setresgid doesn't have a define in macos. used SYS_setreuid and SYS_setregid instead
3. Not a Full solution as under using warp in macos we still get panic on setgid

### CODE REVIEW

**Comment by jackyalbo** [comment]
We can use SYS_setreuid and SYS_setregid instead of SYS_setresuid and SYS_setresgid and then we won't need the special define - I'm not sure if we need to set the saved uid.

File: src/native/fs/fs_napi.cpp [hunk 1]
```diff
@@ -115,29 +138,29 @@ struct FSWorker : public Napi::AsyncWorker
         bool change_uid = orig_uid != _req_uid;
         bool change_gid = orig_gid != _req_gid;
         if (change_gid) {
-            int r = syscall(SYS_setresgid, -1, _req_gid, -1);
+            int r = syscall_3(SYS_SETRESGID, -1, _req_gid, -1);
```
Comment (guymguym) [inline src/native/fs/fs_napi.cpp:141]: isn't it just for Mac? what about linux?

**Comment by guymguym** [comment]
@jackyalbo What did you say happens with the current fix? 
Does it still change the uid/gid for the entire process instead of per thread?

**Comment by jackyalbo** [comment]
> @jackyalbo What did you say happens with the current fix?
> Does it still change the uid/gid for the entire process instead of per thread?

@guymguym  This is my assumption as with current code I get this: ```PANIC: FS::FSWorker::Execute setgid before work failed GID:0|0 UID:30|0```

**Comment by jackyalbo** [comment]
@guymguym I added the thread mutexes - please take another look. Thanks.
The warp run has passed - so it probably was the threads issue in continue to your previous question

**Comment by guymguym** [comment]
I see that there is another way - see `man pthread_setugid_np`.
I opened an alternative PR - #6543

**Pull request closed without merging**